server/odc-service/src/main/java/com/oceanbase/odc/service/iam/JwtService.java

```diff
@@ -0,0 +1,251 @@
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.integration.jdbc.lock.JdbcLockRegistry;
+import org.springframework.stereotype.Component;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTCreator;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.TokenExpiredException;
+import com.auth0.jwt.interfaces.Claim;
+import com.oceanbase.odc.metadb.config.SystemConfigEntity;
+import com.oceanbase.odc.service.config.SystemConfigService;
+import com.oceanbase.odc.service.config.model.Configuration;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Component
+@Slf4j
+public class JwtService {
+
+    @Value("${odc.iam.auth.jwt.expiration-seconds:15*60*1000}")
+    private long expireTime;
+    @Value("${odc.iam.auth.jwt.buffer-seconds:3*60*1000}")
+    private long bufferTime;
+    @Value("${odc.iam.auth.jwt.secret-key:#{null}}")
+    private String tokenSecret;
+    private static final long TRY_LOCK_TIMEOUT_SECONDS = 5;
+
+    private static final String LOCK_KEY = "ODC_JWT_SECRET_LOCK_KEY";
+
+    private static final String ENCRYPTION_JWT_KEY_SYSTEM_CONFIG_KEY = "odc.iam.auth.jwt.secret-key";
+
+    private static final int ENCRYPTION_KEY_SIZE = 64;
+
+    /**
+     * Ensure the consistency of automatically generated jwt key in distributed mode
+     * 
+     * @param tokenSecret
+     * @param systemConfigService
+     * @param jdbcLockRegistry
+     */
+    public JwtService(
+            @Value("${odc.iam.auth.jwt.secret-key:#{null}}") String tokenSecret,
+            SystemConfigService systemConfigService,
+            JdbcLockRegistry jdbcLockRegistry) {
+        try {
+            if (StringUtils.isNotBlank(tokenSecret)) {
+                return;
+            }
+            log.info("Try to lock odc jwt secret...");
+            Lock lock = jdbcLockRegistry.obtain(LOCK_KEY);
+            if (lock.tryLock(TRY_LOCK_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+                try {
+                    log.info("Successfully acquired the jwt secret lock");
+                    List<Configuration> list = systemConfigService.queryByKeyPrefix(
+                            ENCRYPTION_JWT_KEY_SYSTEM_CONFIG_KEY);
+                    if (verifySystemConfig(list)) {
+
+                        this.tokenSecret = list.get(0).getValue();
+                    } else {
+                        this.tokenSecret = UUID.randomUUID().toString();
+                        SystemConfigEntity jwtSecretKey =
+                                createSystemConfigEntity(ENCRYPTION_JWT_KEY_SYSTEM_CONFIG_KEY,
+                                        this.tokenSecret, "ODC jwt secret key");
+                        systemConfigService.insert(Arrays.asList(jwtSecretKey));
```

Comment:
what if the saved tokenSecret value is empty?
the method systemConfigService.insert will make sure use update instead of insert ?

Reply:
I suggest add method for SystemConfigService, such as
1. getConfig
2. saveConfig

Reply:
f the saved tokenSecret value is empty , should update the new tokenSecret . so add saveConfig method. the function of this method is to insert record  when keys don't conflict, and modify value when keys conflict. don't need getConfig, just keeping using the queryByKeyPrefix